torch/lib/THC/THCApply.cuh

```diff
@@ -198,7 +198,10 @@ bool THC_pointwiseApply1(THCState* state,
     TensorInfo<typename TensorUtils<TensorTypeA>::DataType, unsigned int> aInfo =
       getTensorInfo<TensorTypeA, unsigned int>(state, a);
     aInfo.collapseDims();
-
+#if CUDA_VERSION < 9000
+    if (!aInfo.isContiguous())
+        grid.x = THCState_getCurrentDeviceProperties(state)->multiProcessorCount * 4;
+#endif
```

Comment:
shouldn't it be min of current grid.x and numSMs * 4 ? Also, don't really like 4 literal constant - if THC_APPLY_THREADS_PER_BLOCK changes, it would have to change too.